# **Question_2**

In [1]:
import numpy as np
import pandas as pd

In [2]:
data=np.array([108, 138, 124, 163, 124, 159, 106, 134, 115, 139])
#hypothesis mean =120
hyp_mean= 120

# a. Setting up Hypotheses

**Null Hypothesis (H₀): μ ≤ 120 days**

**Alternative Hypothesis (H₁): μ > 120 days**

This is a one-tailed test since we want to demonstrate the mean exceeds 120 days.

In [3]:
#n= number of data points
n= len(data)
#df= degree of freedom
df=n-1
#alpha= significance level
alpha= 0.01

In [4]:
mean= np.mean(data)
std= np.std(data, ddof=1)
mean, std

(np.float64(131.0), np.float64(19.544820285692065))

In [5]:
from scipy.stats import t

In [6]:
#t_stat= t statistics
t_stat = (mean - hyp_mean) / (std / np.sqrt(n))
t_stat

np.float64(1.7797582046490772)

In [7]:
#t_critical= t critical value
t_critical = t.ppf(1 - alpha, df)
t_critical

2.821437925025808


# b. 
Decision Rule: Reject H₀ if t_stat > t_critical

**Since t_stat < t_critical (1.78 < 2.821), we fail to reject H₀**

Conclusion: At α = 0.01, we don't have sufficient evidence to conclude that the mean shelf life exceeds 120 days.

In [8]:
# for one tail test
p_value = 1 - t.cdf(t_stat, df)
p_value

np.float64(0.0544088734750795)

# **c. p_value**= 0.0544(5.44%)

since P_value> alpha(0.01), **Hence we fail to reject H₀.**

# **d**. 
**99 percent confidence interval on the mean shelf life.**

In [9]:
confidence_level = 0.99

#t_conf= t confidence value
#ppf= percent point function
t_conf = t.ppf(1 - (1 - confidence_level) / 2, df)


error = t_conf * (std / np.sqrt(n))
confidence_interval = (mean - error, mean + error)
confidence_interval

(np.float64(110.91401817160775), np.float64(151.08598182839225))

In [10]:
print("data: ", data)
print("hyp_mean: ", hyp_mean)
print("alpha: ", alpha)
print("n: ", n)
print("degree of freedom: ", df)
print("mean: ", mean)
print("standard deviation: ", std)
print("t statistics: ", t_stat)
print("t critical: ", t_critical)
print("p value: ", p_value)
print("confidence interval: ", confidence_interval)
print("error: ", error)
print("t confidence: ", t_conf)

data:  [108 138 124 163 124 159 106 134 115 139]
hyp_mean:  120
alpha:  0.01
n:  10
degree of freedom:  9
mean:  131.0
standard deviation:  19.544820285692065
t statistics:  1.7797582046490772
t critical:  2.821437925025808
p value:  0.0544088734750795
confidence interval:  (np.float64(110.91401817160775), np.float64(151.08598182839225))
error:  20.08598182839225
t confidence:  3.2498355415921254


# **e. Type I and Type II Errors**

**Type I Error (alpha):** Concluding that the mean shelf life exceeds 120 days when it does not.

Probability = 0.01 (given significance level)


**Type II Error (beta):** Failing to conclude that the mean shelf life exceeds 120 days when it actually does.

Probability depends on the true value of the population mean